pkg/fanal/analyzer/pkg/rpm/archive.go

```diff
@@ -136,7 +136,13 @@ func (a *rpmArchiveAnalyzer) generatePURL(pkg *types.Package) *packageurl.Packag
 	case strings.Contains(vendor, "suse"):
 		ns = "suse"
 	}
-	return packageurl.NewPackageURL(packageurl.TypeRPM, ns, pkg.Name, utils.FormatVersion(*pkg), nil, "")
+	return packageurl.NewPackageURL(packageurl.TypeRPM, ns, pkg.Name, utils.FormatVersion(*pkg),
+		packageurl.Qualifiers{
+			{
+				Key:   "arch",
+				Value: pkg.Arch,
```

Comment:
Just to make sure - does the `Arch` field always exist?
I'm worried about the case when `pkg.Arch` is missing. Then we'll see a case like `pkg:rpm/redhat/socat@1.7.3.2-2.el7?arch="`

Reply:
I believe it doesn't happen, but I added [a check](https://github.com/aquasecurity/trivy/pull/7654/commits/5e5f5c14dd71dbfce27e627fd0b17e898206b96e) just in case.